File: java-impl/src/main/java/org/oyster/card/service/CustomerJourney.java
package org.oyster.card.service;

import org.oyster.card.dto.Customer;
import org.oyster.card.dto.JourneyType;
import org.oyster.card.dto.Station;

public interface CustomerJourney {

    void inwardTap(Customer customer, Station station, JourneyType journeyType);

    void outwardTap(Customer customer, Station station);
}

File: kotlin-impl/src/main/kotlin/org/oyster/card/service/CustomerJourney.kt
package org.oyster.card.service

import org.oyster.card.dto.Customer
import org.oyster.card.dto.JourneyType
import org.oyster.card.dto.Station

interface CustomerJourney {

    fun inwardTap(customer : Customer, station: Station, journeyType: JourneyType)

    fun outwardTap(customer: Customer, station: Station)
}
